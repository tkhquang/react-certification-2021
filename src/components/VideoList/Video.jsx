import React from 'react';

import {
  StyledVideo,
  StyledVideoCover,
  StyledVideoTitle,
  StyledVideoDescription,
} from './Video.styled';

export default function Video({ title, description, thumbnails }) {
  return (
    <StyledVideo>
      <StyledVideoCover src={thumbnails.high.url} />
      <StyledVideoTitle>{title}</StyledVideoTitle>
      <StyledVideoDescription>{description}</StyledVideoDescription>
    </StyledVideo>
  );
}
