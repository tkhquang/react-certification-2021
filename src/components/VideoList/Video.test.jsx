import React from 'react';
import renderer from 'react-test-renderer';

import { dummyData } from '../../test/mocks';

import Video from './Video';

describe('Video', () => {
  test('renders with necessary contents', () => {
    const sampleVideoData = dummyData.items[0].snippet;
    const tree = renderer.create(<Video {...sampleVideoData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
