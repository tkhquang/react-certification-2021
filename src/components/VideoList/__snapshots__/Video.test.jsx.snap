// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders with necessary contents 1`] = `
<div
  className="sc-bdnxRM dhSMzM"
>
  <div
    className="sc-gtsrHT jJhotc"
    src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
  />
  <div
    className="sc-dlnjwi hbyQXN"
  >
    Wizeline
  </div>
  <div
    className="sc-hKFxyN jCoFng"
  >
    Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
  </div>
</div>
`;
